openshift/console
# [WIP] Add support to configure the topology icon on the import/edit pages

**Fixes**: 
https://issues.redhat.com/browse/ODC-3847

**Analysis / Root cause**: 
As a user when i deploy an Image, i'd like the opportunity to add a runtime label so that it can be visually identified in the topology view.

**Solution Description**: 
Add a new form field to the import pages to select or enter an icon which will be saved as new label `"app.openshift.io/icon"` (similar to `"app.openshift.io/runtime"`). /cc @christianvogt 

Changed the Topology view so that it prefers the new label and fallback to the runtime label or a default icon if no runtime is specified or unknown.

It uses a Typeahead input field for the supported icons and supports also manual input of URLs. It supports http://, https:// and data: links! Some examples for icon links:

* [ ] todo christoph

**Screen shots / Gifs for design review**: 
@openshift/team-devconsole-ux please check the following (coming soon) screenshots/gifs:

* [ ] todo christoph, I will ping in slack aswell

**Unit test coverage report**: 

* [ ] todo christoph

**Test setup:**
* Open the Add > Import Image page
* Configure the image and create the new Deployment / Deployment Config / Knative
* Check the icon on the Topology Graph
* Edit the icon with right click > "Edit $yourdeployment"
* Check the icon on the Topology Graph, again

**Browser conformance**:
